tests/B2CWebAppCallsWebApi/TodoListService/AuthorizationPolicies/ScopesRequirement.cs

```diff
@@ -28,17 +28,19 @@ public ScopesRequirement(params string[] acceptedScopes)
         protected override Task HandleRequirementAsync(AuthorizationHandlerContext context,
                                                         ScopesRequirement requirement)
         {
+            string scope = "http://schemas.microsoft.com/identity/claims/scope";
+            string scp = "scp";
             // If there are no scopes, do not process
-            if (!context.User.Claims.Any(x => x.Type == ClaimConstants.Scope)
-               && !context.User.Claims.Any(y => y.Type == ClaimConstants.Scp))
+            if (!context.User.Claims.Any(x => x.Type == scope)
```

Comment:
Please use string.equals
Also consider adding static analysis that will prevent you from making this mistake in the future. I recommend you copy the static analysis rules from asp.net core.

Reply:
As far as I remember equality operator for strings calls string.Equals with OrdinalIgnoreCase behind the scenes. Is your suggestion just for convention?

Which static analyzers? Is it the FxCop package or something else?

Reply:
yes. and added an issue for this: [https://github.com/AzureAD/microsoft-identity-web/issues/76](https://github.com/AzureAD/microsoft-identity-web/issues/76)

---
In reply to: [402376930](https://github.com/AzureAD/microsoft-identity-web/pull/75#discussion_r402376930) [](ancestors = 402376930)